Change Makefile.am to install regression tests for test package purpose.

Upstream-Status: Inappropriate [test not install is for purpose from upstream]

Signed-off-by: Yao Zhao <yao.zhao@windriver.com>

diff --git a/test/Makefile.am b/test/Makefile.am
index 7ba11a8..249f0e7 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -13,7 +13,8 @@ else
 TEST_BINARIES=
 endif
 
-noinst_PROGRAMS= $(TEST_BINARIES)
+testdir = $(datadir)/@PACKAGE@/tests
+test_PROGRAMS= $(TEST_BINARIES)
 
 test_service_SOURCES=				\
 	test-service.c
diff --git a/test/interfaces/Makefile.am b/test/interfaces/Makefile.am
index 3cb2c39..e6de67d 100644
--- a/test/interfaces/Makefile.am
+++ b/test/interfaces/Makefile.am
@@ -39,7 +39,8 @@ if DBUS_BUILD_TESTS
 
 ## we use noinst_PROGRAMS not check_PROGRAMS for TESTS so that we
 ## build even when not doing "make check"
-noinst_PROGRAMS = test-service test-client
+testdir = $(datadir)/@PACKAGE@/tests/interfaces
+test_PROGRAMS = test-service test-client
 
 test_service_SOURCES = \
 	test-interfaces.c \
diff --git a/test/core/Makefile.am b/test/core/Makefile.am
index ef6cb26..9786c3a 100644
--- a/test/core/Makefile.am
+++ b/test/core/Makefile.am
@@ -46,7 +46,8 @@ endif
 
 ## we use noinst_PROGRAMS not check_PROGRAMS for TESTS so that we
 ## build even when not doing "make check"
-noinst_PROGRAMS = \
+testdir = $(datadir)/@PACKAGE@/tests/core
+test_PROGRAMS = \
 	test-dbus-glib \
 	test-service-glib \
 	$(THREAD_APPS) \
